sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1314,6 +1314,13 @@ object SQLConf {
       "Other column values can be ignored during parsing even if they are malformed.")
     .booleanConf
     .createWithDefault(true)
+
+  val LEGACY_SIZE_OF_NULL = buildConf("spark.sql.legacy.sizeOfNull")
```

Comment:
`spark.sql.function.sizeOfNull` is better? btw, If major releases happen in Spark, we can remove these kinds of options for back-compatibility? (just a question)

Reply:
We would like to reserve the namespace `spark.sql.legacy.*` for all legacy configurations. /cc @marmbrus @rxin

Reply:
Do we plan to fix other things accordingly too?

Reply:
Yes, we would like to change/improve external behavior under flags in the `spark.sql.legacy.*` namespace. All those flags should be removed in the next major release 3.0. Please, share your thoughts about it.

Reply:
I've created https://issues.apache.org/jira/browse/SPARK-24625 to track it.

It's similar to https://github.com/apache/spark/pull/21427#issuecomment-396142545 , but as I replied in that PR, having version specific config is an overkill, while `legacy` is simpler and more explicit that it will be removed in the future.

Reply:
That's basically the same except that the postfix includes a specific version, which was just a rough idea.